### PR TITLE
feat: force operator deregistration

### PR DESCRIPTION
**Pull request by ypatil12**
Enable operators to force remove themself from operator sets

### CODE REVIEW

**Comment by stevennevins** [comment]
need to update bindings?

**Comment by stevennevins** [comment]
Need to update the base branch to `dev`

File: src/contracts/core/AVSDirectory.sol [hunk 1]
```diff
@@ -197,44 +197,33 @@ contract AVSDirectory is
...
     /**
-     *  @notice Called by AVSs or operators to remove an operator from an operator set.
+     * @notice Called by an operator to deregister from an operator set
+     * 
+     * @param avs The address of the AVS to deregister the operator from.
+     * @param operatorSetIds The IDs of the operator sets.
+     * 
+     * @dev msg.sender used is the operator
+     */
+    function deregisterFromAVSOperatorSets(
```
Comment (stevennevins) [inline src/contracts/core/AVSDirectory.sol:207]: Based on naming discussions going to put some options here:

```
function deregisterSelfFromAVSOperatorSets(
    address avs, 
    uint32[] calldata operatorSetIds
) external;
``` 

```
function leaveAVSOperatorSets(address avs, uint32[] calldata operatorSetIds) external
```

Comment (ypatil12) [reply src/contracts/core/AVSDirectory.sol:207]: I like the first one :)

File: src/contracts/core/AVSDirectory.sol [hunk 2]
```diff
@@ -197,44 +197,33 @@ contract AVSDirectory is
...
      *
      *  @param operator The address of the operator to be removed from the operator set.
      *  @param operatorSetIds The IDs of the operator sets.
      *
      *  @dev msg.sender is used as the AVS.
-     *  @dev The operator must be registered for the msg.sender AVS and the given operator set.
-     *  @dev If this call removes the operator from all operator sets for the msg.sender AVS,
-     *  then an OperatorAVSRegistrationStatusUpdated event is emitted with a DEREGISTERED status.
      */
     function deregisterOperatorFromOperatorSets(
```
[the remark this answers is not in the document]
Comment (ypatil12) [reply src/contracts/core/AVSDirectory.sol:222]: cc @0xClandestine

**Pull request closed without merging**